django/cantusdb_project/main_app/views/chant.py

```diff
@@ -1103,12 +1105,42 @@ def get_chants_with_folios(chants_in_feast):
 
     def form_valid(self, form):
         if form.is_valid():
-            form.instance.last_updated_by = self.request.user
-            messages.success(
-                self.request,
-                "Chant updated successfully!",
-            )
-            return super().form_valid(form)
+            user: User = self.request.user
+            chant: Chant = form.instance
+
+            if not user_can_proofread_chant(user, chant):
+                # Preserve the original values for proofreader-specific fields
+                original_chant: Chant = self.get_object()
+                chant.chant_range = original_chant.chant_range
+                chant.volpiano_proofread = original_chant.volpiano_proofread
+                chant.manuscript_full_text_std_proofread = (
+                    original_chant.manuscript_full_text_std_proofread
+                )
+                chant.manuscript_full_text_proofread = (
+                    original_chant.manuscript_full_text_proofread
+                )
+                proofreaders: list[Optional[User]] = list(
+                    original_chant.proofread_by.all()
+                )
+
+                # Handle proofreader checkboxes
+                if "volpiano" in form.changed_data:
+                    chant.volpiano_proofread = False
+                if "manuscript_full_text_std_spelling" in form.changed_data:
+                    chant.manuscript_full_text_std_proofread = False
+                if "manuscript_full_text" in form.changed_data:
+                    chant.manuscript_full_text_proofread = False
+
+            chant.last_updated_by = user
+            return_response: HttpResponse = super().form_valid(form)
+
+            # The many-to-many `proofread_by` field is reset when the
+            # parent class's `form_valid` method calls `save()` on the model instance.
+            if not user_can_proofread_chant(user, chant):
+                for user in proofreaders:
+                    chant.proofread_by.add(user)
```

Comment:
My only suggestion is that I think this can be simplified like this:

```suggestion
            if not user_can_proofread_chant(user, chant):
                chant.proofread_by.set(user)
```

Reply:
Nice, I've added this change with `chant.proofread_by.set(proofreaders)`

Reply:
Ok! Looks great to me!